subset/switches/src/main/java/switchtest/Main.java

```diff
@@ -4,18 +4,20 @@
 
   public static void main(String[] args) throws Exception {
 
-    if (args.length != 1) {
-      throw new IllegalArgumentException("Expected ipAddress && port as argument");
+    if (args.length != 3) {
+      throw new IllegalArgumentException("Expected ipAddress && port && supportPOE as arguments");
     }
 
     String ipAddress = args[0];
 
     int interfacePort = Integer.parseInt(args[1]);
 
-    SwitchInterrogator switchInterrogator = new SwitchInterrogator(ipAddress, interfacePort);
+    boolean supportsPOE = (args[2].equals("true")) ? true : false;
```

Comment:
@ewRedstone Because supportsPOE is always true (as set up in test_switch) - the PoE part of the test will always be run - what happens if the switch _doesn't_ support PoE? It could be better to _first_ check with `show power-inline` if PoE is enabled or not?

Reply:
I will test out with a poe switch and without a poe switch

Reply:
implemented as suggested first check with a show power-inline command to see if switch supports POE